### PR TITLE
Sends a toast notification when initializations complete.

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
## About The Pull Request

Initialization is significantly slowed down by the presence of clients, though when testing features, you need to join the server. I've been told that some devs (particularly Mothblocks) are alt-tabbed out of Dream Daemon while doing dev work, meaning that they are liable to miss initializations completing, causing an effective slowdown in the dev cycle. Mothblocks said it would be nice if there was a way to produce a desktop notification when initialization completes.

I originally intended to add a function to rust_g that would produce a Windows toast notification with a button allowing you to immediately launch Dream Seeker. However, I couldn't find a reliable way to detect if the OS version was Windows 7 or earlier, so running this function on such an OS would cause a rust panic (which I was told is only a problem because MSO probably still uses Windows 7).

Fortunately, PowerShell scripts can access the necessary .NET APIs to produce toast notifications on Windows 10, while also failing more gracefully than crashing the host process. So I recreated the functionality I intended in PowerShell.

Toast notifications will only be sent on Windows, if the TOAST_NOTIFICATION_ON_INIT config flag is enabled, AND there are no clients on the server.

**Note for downstreams:** If you want the toast notification to have your downstream's icon, copy it, scale the copy down to 16x16, and either rename it "tg_16.png" or change that path in the call to `world.shelleo` to the name of the new file.

Video Demo:

https://user-images.githubusercontent.com/12720844/210492033-963923d7-a1de-4326-9c9f-4f0c0b71d1a5.mp4

## Why It's Good For The Game

This isn't really a line item in the Dev Cycles Initiative, but even if Mothblocks was exaggerating the benefits, it would still be a significant speedup in the dev cycles.

## Changelog

No player-facing changes.

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
Not sure I like this; the idea is good and hilarious, but this seems unnecessary and prone to causing AVs to get mad at an unsigned PS file being ran with ExecutionPolicyBypass

File: code/controllers/configuration/entries/general.dm [hunk 1]
```diff
@@ -645,3 +645,6 @@
 	default = 50
 
 /datum/config_entry/string/morgue_cadaver_override_species
+
+/datum/config_entry/flag/toast_notification_on_init
+	default = FALSE
```
Comment (Mothblocks) [inline code/controllers/configuration/entries/general.dm:650]: i think this its already off by default right

**Comment by Mothblocks** [comment]
uhh is there any way to avoid that cmd popup and have it be a pure daemon

File: code/controllers/configuration/entries/general.dm (hunk 1 above)
Comment (Y0SH1M4S73R) [reply code/controllers/configuration/entries/general.dm:650]: huh, I guess you're right.

File: config/config.txt [hunk 2]
```diff
@@ -632,3 +632,7 @@ PR_ANNOUNCEMENTS_PER_ROUND 5
 
 ## Uncomment to block granting profiling privileges to users with R_DEBUG, for performance purposes
 #FORBID_ADMIN_PROFILING
+
+## Comment to disable sending a toast notification on the host server when initializations complete.
+## Even if this is enabled, a notification will only be sent if there are no clients connected.
+TOAST_NOTIFICATION_ON_INIT
```
Comment (Fikou) [inline config/config.txt:638]: i think you need to comment this to make it disabled by default

Comment (MrStonedOne) [reply config/config.txt:638]: there are two defaults.

code defaults (for if the config doesn't exist in the config file, like on existing servers) and config file defaults, which apply to new copies of the game folder, like those in dev builds.\

They have the config correct for the intended function of being enabled on dev builds but disabled on any prod server that pulls down the changes.

**Comment by Mothblocks** [comment]
If people get annoyed by this we can turn it off by default

**Comment by Mothblocks** [comment]
Hmm now that I have Just Merged it I seem to remember us having problems with shelleo for some reason...? Like it made debugging impossible or something? Uh someone will remind me because I can't find it

**Comment by MrStonedOne** [comment]
> Hmm now that I have Just Merged it I seem to remember us having problems with shelleo for some reason...? Like it made debugging impossible or something? Uh someone will remind me because I can't find it

if the dmb name isn't the same as the parent folder, byond likes to refuse to run worlds in trusted mode or otherwise force the ui prompt

**Comment by MrStonedOne** [comment]
The easiest fix would be to have build tools rename the dmb to be the same as the parent folder. so that tgstation-3/tgstation.m.dmb becomes tgstation-3.dmb (and the same for the rsc) and then byond will be happy

**Comment by Mothblocks** [comment]
I don't think it was that because I have my folder named tgstation but I remember shelleo in auxlua causing problems for me...I'll look into it later

**Pull request merged**